### DISCUSSION TITLE
Xdrip OOP2 libre2 - bluetooth huge battery drain - Half of what the screen uses.

### DISCUSSION THREAD

**Discussion by epifenomen** [NightscoutFoundation/xDrip#3084] [Bug reports] [answered]
xdrip 23ccbd1-2023.07.16 and few other versions like 20230503-e50dfdf, 20230206-ada9236
oop2 27da3f5-2023.05.25 built on: 2123052502
libre2 sensor and librelink (alarms off)
Sony Xperia XZ3 dual sim - Android 10 52.1.A.3.137

I checked different versions, xdrip, except some old ones. The problem has been there from the beginning and has been going on for over a year. When the screen is off, bluetooth consumes 98% of the power. So after the night I have about half of the battery used up.
Actually, everything works fine, except OOP2, which is marked with 5 minutes or an hour or 365 days, and still gives data every minute. Maybe it's OOP2's fault (Unless I can somehow check which app uses BT.). I read somewhere that you can use the old version of Xdrip, which supposedly can get rid of the problem with battery consumption, but I use a modified version of xdrip because the official ones do not accept new sensors from January 2023, I don't know if they changed something in the sensors, but I have to use the version from tzachi-dar with new serial numbers or smth. What can I do to fix this, I can do anything but not change my phone. A one-click solution in Xdrip options would be great :D Amazing work btw. Thank you for xdrip

**Comment by Navid200** [comment]
Please share how you have confirmed that xDrip consumes as much as half of what the screen does.

**Reply by epifenomen** [reply, marked as answer]
I'm very sorry for the confusion, but it turned out that it was my Mi Band 6 with WatchDrip, I installed it together with xDrip and even though I didn't use BT it was eating up the battery. After disconnecting the band from BT, the consumption suddenly dropped to zero. I wouldn't have found it if it weren't for the GSam Battery app. Now when you asked for confirmation to take a screenshot of the comparison, I noticed there was no difference, even after killing the processes. Once again, I apologize to everyone and to the battery of my poor smartphone, which I killed almost in the last year :(
